packages/renderer/src/lib/ui/TerminalWindows.spec.ts

```diff
@@ -0,0 +1,162 @@
+/**********************************************************************
+ * Copyright (C) 2025 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import '@testing-library/jest-dom/vitest';
+
+import { render } from '@testing-library/svelte';
+import { FitAddon } from '@xterm/addon-fit';
+import { Terminal } from '@xterm/xterm';
+import { writable } from 'svelte/store';
+import { beforeEach, expect, test, vi } from 'vitest';
+
+import TerminalWindow from '/@/lib/ui/TerminalWindow.svelte';
+
+import { TerminalSettings } from '../../../../main/src/plugin/terminal-settings';
+
+vi.mock('@xterm/xterm', () => ({
+  Terminal: vi.fn(),
+}));
+
+vi.mock('@xterm/addon-fit', () => ({
+  FitAddon: vi.fn(),
+}));
+
+const TerminalMock: Terminal = {
+  onWriteParsed: vi.fn(),
+  onResize: vi.fn(),
+  dispose: vi.fn(),
+  loadAddon: vi.fn(),
+  open: vi.fn(),
+  write: vi.fn(),
+} as unknown as Terminal;
+
+const FitAddonMock: FitAddon = {
+  fit: vi.fn(),
+} as unknown as FitAddon;
```

Comment:
question: I'm wondering why TerminalMock definition is not part of the vi.mock('@xterm/xterm') definition and FitAddonMock in @xterm/addon-fit

and the pattern
```
   vi.mocked(Terminal).mockReturnValue(TerminalMock);
   vi.mocked(FitAddon).mockReturnValue(FitAddonMock);
```

seems to not fit in an existing test pattern.

Reply:
If I do the following 

```ts
vi.mock('@xterm/addon-fit', () => ({
  FitAddon: vi.fn().mockReturnedValue({
      fit: vi.fn(),
  }),
}));
```

The mockReturnedValue is reset if I have `vi.resetAllmock()` in the `beforeEach`. So I have to define everytime I reset it 🤔 

How do you mock the constructor then and access the object created ?

Reply:
Following our discussion, found a nice way to simplify all the tests: everyone happy!